### PR TITLE
Fix the recovery tests under mac os x

### PR DESCRIPTION
**Pull request by hkernbach**
This fix addresses: https://github.com/arangodb/release-3.4/issues/5

### CODE REVIEW

**Comment by hkernbach** [comment]
http://jenkins01.arangodb.biz:8080/view/Nightlies/job/arangodb-3.4-mac-matrix-nightly/23/

File: lib/Basics/files.cpp [hunk 1]
```diff
@@ -1150,7 +1150,6 @@ int TRI_VerifyLockFile(char const* filename) {
     LOG_TOPIC(WARN, arangodb::Logger::FIXME) << "found existing lockfile '" << filename << "' of previous process with pid " << pid << ", and that process seems to be still running";
   }
 
-#ifdef TRI_HAVE_SETLK
```
Comment (graetzer) [inline lib/Basics/files.cpp]: is `F_SETLK` available on windows ?

Comment (graetzer) [reply lib/Basics/files.cpp]: I think you should probably rather add the define for MacOs in `lib/Basics/operating-system.h:` alternatively remove it completely

**Comment by hkernbach** [comment]
Next try: http://jenkins01.arangodb.biz:8080/view/Nightlies/job/arangodb-3.4-mac-matrix-nightly/26/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1263/

**Pull request merged**